cmd/auth-rpc-client.go

```diff
@@ -146,21 +142,34 @@ func (authClient *AuthRPCClient) Call(serviceMethod string, args interface {
 	SetToken(token string)
 	SetTimestamp(tstamp time.Time)
 }, reply interface{}) (err error) {
-	// On successful login, attempt the call.
-	if err = authClient.Login(); err == nil {
-		// Set token and timestamp before the rpc call.
-		args.SetToken(authClient.serverToken)
-		args.SetTimestamp(time.Now().UTC())
-
-		// Call the underlying rpc.
-		err = authClient.rpc.Call(serviceMethod, args, reply)
-
-		// Invalidate token, and mark it for re-login on subsequent reconnect.
-		if err == rpc.ErrShutdown {
-			authClient.mu.Lock()
-			authClient.isLoggedIn = false
-			authClient.mu.Unlock()
+	loginAndCallFn := func() error {
+		// On successful login, proceed to attempt the requested service method.
+		if err = authClient.Login(); err == nil {
+			// Set token and timestamp before the rpc call.
+			args.SetToken(authClient.serverToken)
+			args.SetTimestamp(time.Now().UTC())
+
+			// Finally make the network call using net/rpc client.
+			err = authClient.rpc.Call(serviceMethod, args, reply)
 		}
+		return err
+	}
```

Comment:
`gofmt` is a tool to make formats for syntax beautification. It doesn't do logic separation/beautification of the code blocks.  That's why `gofmt` doesn't worry about single newline within code blocks.

If code is not logically separated, its hard to read/understand.